src/main/java/aeneas/views/PlayLevelView.java

```diff
@@ -62,19 +79,75 @@
 
   @Override
   public void initialize(URL location, ResourceBundle resources) {
-    bullpenView = new BullpenView(model, bullpenBox, (Pane) this);
+    bullpenView = new BullpenView(model,levelModel.getBullpen(), bullpenBox, (Pane) this);
 
     resetLevelButton.setOnMouseClicked((e) -> {
-      levelModel.reset();
+      this.levelModel.reset();
+      refresh();
     });
 
-    bullpenView.refresh(levelModel.getBullpen());
+    bullpenView.refresh();
+    levelLabel.setText("Level "+levelModel.getLevelNumber());
+
+    levelTypeIcon.setGlyphName(levelModel.getIconName());
 
     boardView = new BoardView(this, model, levelModel.getBoard());
+    boardView.setRefreshListener(this);
 
     VBox.setMargin(boardView, new Insets(10, 10, 10, 10));
     centerBox.setAlignment(Pos.TOP_RIGHT);
     centerBox.getChildren().add(boardView);
+
+    model.setActiveLevel(levelModel);
+    levelModel.start();
+    refresh();
+
+    if (timer != null) timer.cancel();
+    timer = new Timer();
+    timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        Platform.runLater(() -> refresh());
+      }
+    }, 1000, 1000);
+  }
+
+  public void cleanup() {
+    timer.cancel();
+  }
+
+  public void refresh() {
+    int stars = levelModel.getStarsEarned();
+    switch(stars) {
+    case 0:
+      star1.setGlyphName("STAR_ALT");
+      star2.setGlyphName("STAR_ALT");
+      star3.setGlyphName("STAR_ALT");
+      break;
+    case 1:
+      star1.setGlyphName("STAR");
+      star2.setGlyphName("STAR_ALT");
+      star3.setGlyphName("STAR_ALT");
+      break;
+    case 2:
+      star1.setGlyphName("STAR");
+      star2.setGlyphName("STAR");
+      star3.setGlyphName("STAR_ALT");
+      break;
+    case 3:
+      star1.setGlyphName("STAR");
+      star2.setGlyphName("STAR");
+      star3.setGlyphName("STAR");
+      break;
+    }
+    model.updateStats();
+    bullpenView.refresh();
+    boardView.refresh();
+    timeLabel.setText(levelModel.getCountdownText());
+    if (levelModel.isFinished() || stars == 3) {
+      cleanup();
+      mainView.switchToPlaySelectLevelView();
```

Comment:
This causes a problem with the navigation stack (When a level ends and it goes back to the select level view, if you hit back from there, it'll bring you into the level view instead of the main menu). I think we probably want to navigate back instead of explicitly switching to the select level view.